PR: https://github.com/odoo/odoo/pull/217987

From: 1d30ecfc5681bdf51a60f6447a13415d4142932c
From: Stéphane Diez
Date: 2025-09-04 15:16:28

Breaking data model changes scores: del:3 + add:0, change matches:
-    manufacturing_lead = fields.Float(
-    manufacturing_lead = fields.Float(related='company_id.manufacturing_lead', string="Manufacturing Lead Time", readonly=False)
-    use_manufacturing_lead = fields.Boolean(string="Default Manufacturing Lead Time", config_parameter='mrp.use_manufacturing_lead')

Total Changes: 21

[REF] mrp_*, purchase_*: remove security lead time

We are removing po_lead and manufacturing_lead. Nothing is replacing them but
the user can still use `bom.produce_delay`, `supplierinfo.delay` and
`res_company.days_to_purchase` to apply lead times as needed.

task 4779169

Part-of: odoo/odoo#217987
Related: odoo/enterprise#90534
Related: odoo/upgrade#8192
Signed-off-by: Tiffany Chang (tic) <tic@odoo.com>

================================= pseudo patch: =================================

--- a/addons/mrp/models/mrp_bom.py
+++ b/addons/mrp/models/mrp_bom.py
@@ -86,8 +86,7 @@ class MrpBom(models.Model):
         help="Average lead time in days to manufacture this product. In the case of multi-level BOM, the manufacturing lead times of the components will be added. In case the product is subcontracted, this can be used to determine the date at which components should be sent to the subcontractor.")
     days_to_prepare_mo = fields.Integer(
         string="Days to prepare Manufacturing Order", default=0,
-        help="Create and confirm Manufacturing Orders this many days in advance, to have enough time to replenish components or manufacture semi-finished products.\n"
-             "Note that security lead times will also be considered when appropriate.")
+        help="Create and confirm Manufacturing Orders this many days in advance, to have enough time to replenish components or manufacture semi-finished products.")
 
     _qty_positive = models.Constraint(
         'check (product_qty > 0)',

--- a/addons/mrp/models/res_company.py
+++ b/addons/mrp/models/res_company.py
@@ -7,10 +7,6 @@ from odoo import api, fields, models
 class ResCompany(models.Model):
     _inherit = 'res.company'
 
-    manufacturing_lead = fields.Float(
-        'Manufacturing Lead Time', default=0.0, required=True,
-        help="Security days for each manufacturing operation.")
-
     def _create_unbuild_sequence(self):
         unbuild_vals = []
         for company in self:

--- a/addons/mrp/models/res_config_settings.py
+++ b/addons/mrp/models/res_config_settings.py
@@ -7,8 +7,6 @@ from odoo import api, fields, models
 class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
-    manufacturing_lead = fields.Float(related='company_id.manufacturing_lead', string="Manufacturing Lead Time", readonly=False)
-    use_manufacturing_lead = fields.Boolean(string="Default Manufacturing Lead Time", config_parameter='mrp.use_manufacturing_lead')
     group_mrp_byproducts = fields.Boolean("By-Products",
         implied_group='mrp.group_mrp_byproducts')
     module_mrp_mps = fields.Boolean("Master Production Schedule")
@@ -37,11 +35,6 @@ class ResConfigSettings(models.TransientModel):
             # Disabling this option should not interfere with currently planned productions
             self.env['mrp.bom'].sudo().search([('allow_operation_dependencies', '=', True)]).allow_operation_dependencies = False
 
-    @api.onchange('use_manufacturing_lead')
-    def _onchange_use_manufacturing_lead(self):
-        if not self.use_manufacturing_lead:
-            self.manufacturing_lead = 0.0
-
     @api.onchange('group_unlocked_by_default')
     def _onchange_group_unlocked_by_default(self):
         """ When changing this setting, we want existing MOs to automatically update to match setting. """

--- a/addons/mrp/models/stock_rule.py
+++ b/addons/mrp/models/stock_rule.py
@@ -222,13 +222,6 @@ class StockRule(models.Model):
                     for key, value in extra_delays.items():
                         delays[key] += value
                     delay_description += extra_delay_description
-            # manufacturing security lead time
-            for comp in self.picking_type_id.company_id:
-                security_delay = comp.manufacturing_lead
-                delays['total_delay'] += security_delay
-                delays['security_lead_days'] += security_delay
-            if not bypass_delay_description:
-                delay_description.append((_('Manufacture Security Lead Time'), _('+ %d day(s)', security_delay)))
         days_to_order = values.get('days_to_order', bom.days_to_prepare_mo)
         delays['total_delay'] += days_to_order
         if not bypass_delay_description:
